plugin/src/main/java/io/fabric8/maven/plugin/BuildMojo.java

```diff
@@ -241,7 +293,191 @@ private void executeOpenShiftBuild(ServiceHub hub, ImageConfiguration imageConfi
+                if (isBuildCompleted(action, resource)) {
+                    latch.countDown();
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException cause) {
+            }
+        };
+        log.info("Waiting for build " + buildName + " to complete...");
+        try (Watch watcher = client.builds().withName(buildName).watch(buildWatcher)) {
+            while (latch.getCount() > 0L) {
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    // ignore
+                }
+            }
+            log.info("Build " + buildName + " completed!");
+        }
+    }
+
+
+    /**
+     * Lets update the ImageStream to include the correct tags
+     */
+    private void updateImageStreamTags(OpenShiftClient client, ImageConfiguration imageConfig, String buildConfigName, Build build) throws MojoExecutionException {
+
+        try {
+            File manifest = openshiftManifest;
+            if (!Files.isFile(manifest)) {
+                throw new MojoFailureException("No such generated manifest file: " + manifest);
+            }
+
+            String namespace = clusterAccess.getNamespace();
+            Controller controller = new Controller(client);
+
+            boolean updated = false;
+            Set<HasMetadata> entities = loadResources(client, controller, namespace, manifest, project, log);
+            for (HasMetadata entity : entities) {
+                if (entity instanceof ImageStream) {
+                    ImageStream is = (ImageStream) entity;
+                    String imageStreamName = KubernetesHelper.getName(is);
+                    if (Objects.equals(buildConfigName, imageStreamName)) {
+                        if (updateImageStreamTag(client, imageConfig, is, buildConfigName)) {
+                            updated = true;
+                        }
+                    }
+                }
+            }
+            if (updated) {
+                // lets store the entities again!
+                KubernetesList entity = new KubernetesListBuilder().withItems(new ArrayList<>(entities)).build();
+                File resourceFileBase = new File(this.targetDir, ResourceClassifier.OPENSHIFT.getValue());
+                AbstractResourceMojo.writeResourcesIndividualAndComposite(entity, resourceFileBase, ResourceFileType.yaml, log);
+            }
+        } catch (KubernetesClientException e) {
+            KubernetesResourceUtil.handleKubernetesClientException(e, this.log);
+        } catch (MojoExecutionException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+    }
+
+    private boolean updateImageStreamTag(OpenShiftClient client, ImageConfiguration imageConfig, ImageStream is, String buildConfigName) throws MojoExecutionException {
```

Comment:
would call it differently since you are updating an `ImageStream` , not an `ImageStreamTag`.